edge-modules/ModuleRestarter/src/Settings.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+    public class Settings
+    {
+        static readonly Lazy<Settings> DefaultSettings = new Lazy<Settings>(
+            () =>
+            {
+                IConfiguration configuration = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("config/settings.json", optional: true)
+                    .AddEnvironmentVariables()
+                    .Build();
+
+                return new Settings(
+                    configuration.GetValue<string>("ServiceClientConnectionString"),
+                    configuration.GetValue<string>("IOTEDGE_DEVICEID"),
+                    configuration.GetValue<string>("DesiredModulesToRestartCSV"),
+                    configuration.GetValue<int>("RestartIntervalInMins", 10));
+            });
+
+        Settings(
+            string serviceClientConnectionString,
+            string deviceId,
+            string desiredModulesToRestartCSV,
+            int restartIntervalInMins)
+        {
+            this.ServiceClientConnectionString = Preconditions.CheckNonWhiteSpace(serviceClientConnectionString, "ServiceClientConnectionString");
+            this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, "DeviceId");
+            this.DesiredModulesToRestartCSV = Preconditions.CheckNonWhiteSpace(desiredModulesToRestartCSV, "DesiredModulesToRestartCSV");
+            this.RestartIntervalInMins = Preconditions.CheckRange(restartIntervalInMins, 0);
+        }
+
+        public static Settings Current => DefaultSettings.Value;
+
+        public string ServiceClientConnectionString { get; }
+
+        public string DeviceId { get; }
+
+        private string DesiredModulesToRestartCSV { get; }
```

Comment:
I chose to keep this string field and make private to avoid using new in constructor to initialize a list.

Reply:
why do you need this field anymore? you can just add List<string> DesiredModulesToRestart {get;}.

Reply:
Settings takes in a string so we could use a list for this field, however would have to initialize it in the constructor via new(), which I thought was a bad practice.